### PR TITLE
[PT FE] Add aten::logcumsumexp

### PR DESCRIPTION
**Pull request by VectorNd**
Added aten::logcumsumexp

### CODE REVIEW

File: src/frontends/pytorch/src/op/logcumsumexp.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    auto dim = context.get_input(1);
+
+    // First compute exp(input)
+    auto exp = context.mark_node(std::make_shared<v0::Exp>(input));
+    
+    // Then compute cumsum of the exponentials
+    auto cumsum = context.mark_node(std::make_shared<v0::CumSum>(exp, dim));
+    
+    // Finally take log of the result
+    auto log = context.mark_node(std::make_shared<v0::Log>(cumsum));
```
Comment (rkazants) [inline src/frontends/pytorch/src/op/logcumsumexp.cpp:32]: we may have exponent explosion for large `input` elements during computation of exp^(input_el).
In order to avoid this, please do the following:
1. compute max elements along the required dim. `max=ReduceMax(input, dim)`
2. `ln(cumsum(exp(input), dim)) = ln(exp(max) cumsum(exp(input-max), dim)) = max + ln(cumsum(exp(input-max), dim))`

File: tests/layer_tests/pytorch_tests/test_logcumsumexp.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            def forward(self, x):
+                return torch.logcumsumexp(x, self.axis)
+
+        ref_net = None
+
+        return aten_logcumsumexp(axis), ref_net, "aten::logcumsumexp"
+
+    @pytest.mark.parametrize("axis", [0, 1, 2, 3, -1, -2, -3, -4])
+    @pytest.mark.nightly
+    @pytest.mark.precommit
```
Comment (rkazants) [inline tests/layer_tests/pytorch_tests/test_logcumsumexp.py:29]: please also enable this operation for `torch.export`. Need to add `@pytest.mark.precommit_torch_export`

File: src/frontends/pytorch/src/op/logcumsumexp.cpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    auto max = context.mark_node(std::make_shared<v0::ReduceMax>(input,dim))
+
+    // First compute exp(input)
+    auto exp = context.mark_node(std::make_shared<v0::Exp>(input-max));
+    
+    // Then compute cumsum of the exponentials
+    auto cumsum = context.mark_node(std::make_shared<v0::CumSum>(exp, dim));
+    
+    // Finally take log of the result
+    auto log = max + context.mark_node(std::make_shared<v0::Log>(cumsum));
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/logcumsumexp.cpp:34]: there is no `+` operation for nodes, use `Add` operation

**Review by mvafin — CHANGES_REQUESTED** [review]

File: src/frontends/pytorch/src/op/logcumsumexp.cpp [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+using namespace ov::op;
+
+OutputVector translate_logcumsumexp(const NodeContext& context) {
+    // aten::logcumsumexp(Tensor self, int dim) -> Tensor
+    num_inputs_check(context, 2, 2);
+    auto input = context.get_input(0);
+    auto dim = context.get_input(1);
+
+    auto max = context.mark_node(std::make_shared<v0::ReduceMax>(input,dim))
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/logcumsumexp.cpp:25]: Please add necessary includes. Your code cannot be built

**Comment by mlukasze** [comment]
hey @VectorNd any progress here? :)
